public/js/reducers/sidebars.js

```diff
@@ -0,0 +1,69 @@
+// @flow
+const I = require("immutable");
+const makeRecord = require("../utils/makeRecord");
+const C = require("../constants");
+
+import type { Action } from "../actions/types";
+import type { Record } from "../utils/makeRecord";
+
+type SidebarState = {
+  collapsed: boolean,
+  width: number,
+  prevWidth: number,
+};
+
+type SidebarsState = {
+  left: I.Map<string, SidebarState>,
+  right: I.Map<string, SidebarState>,
+};
+
+const State = makeRecord(({
+  left: I.Map({
+    collapsed: false,
+    width: 300,
+    prevWidth: 0
```

Comment:
Ah yeah, if I pass collapsed into splitbox this works wonderfully, `prevWidth` was only nice  because splitbox only uses `width` currently.